aspnetcore/fundamentals/minimal-apis/aspnetcore-openapi.md

```diff
@@ -0,0 +1,322 @@
+ms.custom: mvc
+ms.date: 05/06/2024
+uid: fundamentals/minimal-apis/aspnetcore-openapi
+---
+# Get started with Microsoft.AspNetCore.OpenApi
+
+The Microsoft.AspNetCore.OpenApi provides built-in support for OpenAPI document generation in ASP.NET Core. The package is:
+
+* Compatible with native AoT
+* Takes advantage of JSON schema support provided by System.Text.Json
+* Provides a transformers API for modifying generated documents
+* Supports managing multiple OpenAPI documents within a single application
+
+## Package installation
+
+`Microsoft.AspNetCore.OpenApi` can be added with the following approaches:
+
+### [Visual Studio](#tab/visual-studio)
+
+* From the **Package Manager Console** window:
+  * Go to **View** > **Other Windows** > **Package Manager Console**
+  * Navigate to the directory in which the `.csproj` file exists
+  * Execute the following command:
+
+    ```powershell
+    Install-Package Microsoft.AspNetCore.OpenApi -Version 9.0.0
+    ```
+
+* From the **Manage NuGet Packages** dialog:
+  * Right-click the project in **Solution Explorer** > **Manage NuGet Packages**
+  * Set the **Package source** to "nuget.org"
+  * Ensure the "Include prerelease" option is enabled
+  * Enter "Microsoft.AspNetCore.OpenApi" in the search box
+  * Select the latest "Microsoft.AspNetCore.OpenApi" package from the **Browse** tab and click **Install**
+
+### [Visual Studio Code](#tab/visual-studio-code)
+
+Run the following command from the **Integrated Terminal**:
+
+```dotnetcli
+dotnet add package Microsoft.AspNetCore.OpenApi -v 9.0.0
+```
+
+### [.NET Core CLI](#tab/netcore-cli)
+
+Run the following command:
+
+```dotnetcli
+dotnet add package Microsoft.AspNetCore.OpenApi -v 9.0.0
+```
+
+---
+
+## Add and configure OpenAPI document generation
+
+To get started generating OpenAPI documents for ASP.NET Core applications, add the OpenAPI-related services to `Program.cs`.
+
+```csharp
+builder.Services.AddOpenApi();
+```
+
+Then, enable the endpoint for viewing the OpenAPI document in JSON format.
+
+```csharp
+app.MapOpenApi();
+```
+
+Launch the app and navigate to `https://localhost:<port>/openapi/v1.json` to view the generated OpenAPI document.
+
+## Options for customizing OpenAPI document generation
+
+### Customizing OpenAPI version of generated document
+
+By default, OpenAPI document generation will generate a document that is compliant with v3.0 of the OpenAPI spec. To modify this version, customize the options that are provided to the document generation services.
+
+```csharp
+builder.Services.AddOpenApi(options =>
+{
+    options.OpenApiVersion = OpenApiSpecVersion.OpenApi2_0;
+});
+```
+
+### Customizing the OpenAPI endpoint route
+
+By default, the OpenAPI endpoint registered via a call to `MapOpenApi` will expose the document at the `/openapi/{documentName}.json` endpoint. To customize the route the OpenAPI document is registered at, pass the route template as a parameter to the `MapOpenApi` call.
```

Comment:
Where does the value of `documentName` come from? Can we explain that here?

Reply:
Oh ... I see ... it is described below. Still, I think worth pointing to that section -- or switching the order so that folks have the context.

Reply:
Good point -- I'll move document name higher up since it's a more important topic.